src/arkode/arkode.c

```diff
@@ -1969,17 +2037,30 @@ int arkInitialSetup(ARKodeMem ark_mem, sunrealtype tout)
     /* Estimate initial h if not set */
     if (ark_mem->h == ZERO)
     {
-      /* Again, temporarily set h for estimating an optimal value */
+      /* If necessary, temporarily set h as it is used to compute the tolerance
+         in a potential mass matrix solve when computing the full rhs */
       ark_mem->h = SUNRabs(tout - ark_mem->tcur);
-      if (ark_mem->h == ZERO) { ark_mem->h = ONE; }
+      if (ark_mem->h == ZERO) ark_mem->h = ONE;
+
+      /* Call fullrhs if needed */
+      if (!(ark_mem->fn_is_current))
+      {
+        retval = ark_mem->step_fullrhs(ark_mem, ark_mem->tcur, ark_mem->yn,
```

Comment:
I did this when figuring out how to estimate the initial step sizes for multirate calculations.  It's been about 18 months since I implemented this so my memory of the details are murky, but I believe that if `step_fullrhs` remained inside `arkHin` I ended up having to evaluate the RHS one too many times.

Reply:
I believe that @gardner48 is correct.

Reply:
I don't remember seeing this reuse in going over the code so it might be that this can stay inside `arkHin`